src/components/chooseFolder/chooseFolder.js

```diff
@@ -32,11 +33,10 @@ class ChooseFolder extends Component {
   	}
 
   	render() {
-       	const listOfFileNames = [];
+		   const listOfFileNames = [];
        	for(var key of this.state.path) {
-   	        listOfFileNames.push(<CheckBoxSelection key={key.name} filename={key.name} />);
-       	}
-	
+   	        listOfFileNames.push(<CheckBoxSelection key={key.name} filename={key.name} file={key} />);
+		   }	
```

Comment:
please use array.map here instead of the for and push

Reply:
This is because .map will produce a new array and push mutates the array data. Best practice for react is to use immutable data as much as you can. Because of the nature of how react works, mutation of data objects can cause serious and difficult to debug errors

Reply:
Thanks for knowledge,  I will attend to it accordingly.

Reply:
I am still struggling with this one its giving me errors currently

Reply:
remember, map returns an array, so you have to assign it to a variable

Reply:
Oh I see where the error is coming,  I am using an array method  on an object. like map is expecting [] but its finding { }, there must be a way to use map in an object.   thanks

Reply:
This is my approach `const listOfFileNames = Object.keys(this.state.path).map(function(key) {
			<div><CheckBoxSelection key={this.state.path[key].name} filename={this.state.path[key].name} file={key} /></div>
		  });` and it wont works complaining with `Expected an assignment or function call and instead saw an expression`